compiler/z/codegen/OMRTreeEvaluator.cpp

```diff
@@ -3952,31 +3952,39 @@ genCompareAndBranchInstructionIfPossible(TR::CodeGenerator * cg, TR::Node * node
    return cursor;
    }
 
-TR::InstOpCode::S390BranchCondition useBranchOnZeroConditionMaskIfPossible(TR::Node * node, TR::InstOpCode::S390BranchCondition brCond)
+/**
+ * \fn convertComparisonBranchConditionToTestUnderMaskBranchCondition(TR::Node * node, TR::InstOpCode::S390BranchCondition brCond)
+ * \brief Converts a branch condition mask to the correct mask for the test under mask instruction being generated 
+ * 
+ * \param node Constant, 1st or 2nd child of IL tree root
+ * \param brCond Branch condition mask 
+ * \return newCond New mask to be used in generated test under mask instruction 
+ *
+ */
+TR::InstOpCode::S390BranchCondition convertComparisonBranchConditionToTestUnderMaskBranchCondition(TR::Node * node, TR::InstOpCode::S390BranchCondition brCond)
```

Comment:
This function returns a branch condition, not a mask. Would renaming `node` to `constNode` be more intuitive? The comment also seems to imply restrictions on where `node` comes from, i.e. first or second child. This restriction does not make sense. How about:

```
/** \brief
 *     Converts a comparison branch condition to the correct branch condition for the test under mask instruction 
 *     being generated.
 *
 *  \param constNode
 *     The constant the comparison branch condition is comparing against.
 *
 *  \param brCond
 *     The comparison branch condition to be converted.
 *
 *  \return
 *     The branch condition to be used in generated test under mask instruction.
 */
```

Regardless the comment seems to imply the consumer of this function must pass in a constant. Lets enforce this invariant with an assert at the start of the function.